docs/usage/shoot_updates.md

```diff
@@ -0,0 +1,93 @@
+
+### Which Updates are Applied
+
+As end-users can only control the `Shoot` resource's specification but not the used Gardener version, they don't have any influence on which of the updates are rolled out (other than those settings configurable in the `Shoot`).
+A Gardener operator can deploy a new Gardener version at any point in time.
+Any subsequent reconciliation of `Shoot`s will update them by rolling out the changes incorporated in this new Gardener version.
+
+Some examples for such shoot updates are:
+
+* Add a new/remove an old component to/from the shoot's control plane running in the seed, or to/from the shoot's system components running on the worker nodes.
+* Change the configuration of an existing control plane/system component.
+* Add a new/remove an old condition to/from the `.status.conditions[]` list maintained in the `Shoot` resource.
+* Restart of existing control plane/system components (this might result in a short unavailability of the Kubernetes API server, e.g., when etcd or a kube-apiserver itself is being restart)
+
+### Behavioural Changes
+
+Generally, some of such updates (e.g., configuration changes) could theoretically result in different behaviour of controllers.
+If such changes would be backwards-incompatible then we usually follow one of those approaches (depends on the concrete change):
+
+* Only apply the change for new clusters.
+* Expose a new field in the `Shoot` resource that lets users control this changed behaviour to enable it at a convenient point in time.
+* Put the change behind an alpha feature gate (disabled by default) in the gardenlet (only controllable by Gardener operators) which will be promoted to beta (enabled by default) in subsequent releases (in this case, end-users have no influence on when the behaviour changes - Gardener operators should inform their end-users and provide clear timelines when they will enable the feature gate).
+
+## Upgrades
+
+We consider shoot upgrades to change either the
+
+* Kubernetes version (`.spec.kubernetes.version`)
+* Machine image version of at least one worker pool (`.spec.provider.workers[].machine.image.version`)
+
+Generally, an upgrade is also performed through a reconciliation of the `Shoot` resource, i.e., the same concepts like for [shoot updates](#updates) apply.
+If an end-user triggers an upgrade (e.g., by changing the Kubernetes version) after a new Gardener version was deployed but before the shoot was reconciled again, then this upgrade might incorporate the changes delivered with this new Gardener version.
+
+### In-Place vs. Rolling Updates
+
+If the Kubernetes patch version is changed then the upgrade happens in-place.
+This means that the shoot worker nodes remain untouched and only the `kubelet` process restarts with the new Kubernetes version binary.
```

Comment:
What about changes to the cri or kubelet config? What about labels and taints?
They get updated in-place, right? I think it makes sense to specify that here, similar to the section `Rolling Update Triggers ` down below.

Reply:
I added a comment about the kubelet configuration. However, we already maintain the list of fields that lead to a rolling update. Out of this list you already know that all other changes will be done in-place. Should be good enough, or?

Reply:
Yep, right. This should be enough :)